guides/duplicate-theming-styles.md

```diff
@@ -0,0 +1,92 @@
+# Avoiding duplicated theming styles
+
+As explained in the [theming guide](./theming.md), a theme in Angular Material consists of
+configurations for the `color`, `density` and `typography` systems. As some of these individual
+systems have default configurations, some usage patterns may cause duplication in the CSS output.
+
+Below are examples of patterns that generate duplicative theme styles:
+
+**Example #1**
+
+```scss
+$light-theme: mat-light-theme((color: ...));
+$dark-theme: mat-dark-theme((color: ...));
+
+// Generates styles for all systems configured in the theme. In this case, color styles
+// and default density styles are generated. Density is in themes by default.
+@include angular-material-theme($light-theme);
+
+.dark-theme {
+  // Generates styles for all systems configured in the theme. In this case, color styles
+  // and the default density styles are generated. **Note** that this is a problem because it
+  // means that density styles are generated *again*, even though only the color should change.
+  @include angular-material-theme($dark-theme);
+}
+```
+
+To fix this, you can use the dedicated mixin for color styles for the `.dark-theme`
+selector. Replace the `angular-material-theme` mixin and include the dark theme using the
+`angular-material-color` mixin. For example:
+
+```scss
+...
+@include angular-material-theme($light-theme);
+
+.dark-theme {
+  // This mixin only generates the color styles now.
+  @include angular-material-color($dark-theme);
+}
+```
+
+Typography can also be configured via Sass mixin; see `angular-material-typography`.
+
+**Example #2**
+
+Theme styles could also be duplicated if individual theme mixins are used. For example:
+
+```scss
+@include angular-material-theme($my-theme);
+
+.my-custom-dark-button {
+  // This will also generate the default density styles again.
+  @include mat-button-theme($my-theme);
+}
+```
+
+To avoid this duplication of styles, use the dedicated mixin for the color system and
+extract the configuration for the color system from the theme.
+
+```scss
+.my-custom-dark-button {
+  // This will only generate the color styles for `mat-button`.
+  @include mat-button-color(map_get($my-theme, color));
```

Comment:
This method seems simpler, I think we should just recommend this and remove the alternative

Reply:
Agreed. Done.

Reply:
This might be a followup for another PR, but should we make it so that if you pass a full theme to the mat-*-color mixins it just does the `map_get($theme, color)` for you? Seems like it would make the user's life easier

Reply:
That something I think we talked about in our sync meeting already. We decided to do it only for the overall all-* mixins, as otherwise we'd need to add the theme unwrapping to every `-color`, `-density` and `-typography` mixin. 

I was thinking more about this too. The current API requires developers to really understand how our themes are constructed (i.e. map w/ keys for each sub-system). I'm unsure about that. I would definitely like the convenience API for consumers, but at the same time it feels like a _lot_ of code/churn for us. Maybe we can just expect this concept to be understood?

Reply:
Prior to these changes it feels like we've gone out of our way to make the structure of these objects pretty opaque and just provide mixins and functions that know how to create and use them (e.g. `mat-light-theme`, `mat-color`, etc). This is the first time we're telling people to use `map_get` directly in their code, so it feels like a departure from what we've been doing...

Reply:
Yeah, that's what I'm saying. I agree with you that this is more inconvenient. Though it's also unclear to me whether we want to go down that road and handle it for all the individual mixins. 

Alternatively we can just recommend using another method like `mat-color-config($theme)`

Reply:
or make `mat-get` an alias for `map_get` XD

Reply:
hahah, that's a good one 😄